### DISCUSSION TITLE
single crystal superalloys

### DISCUSSION THREAD

**Discussion by liangcl123** [neperfepx/FEPX#117] [Q&A]
I am a beginner in FEPX. Can FEPX be used to simulate single crystal superalloys. I know that FEPX can simulate multiphase polycrystals, but I am not sure if it can solve the simulation problem of single crystals. Please guide me. Thank you!

**Comment by mkasemer** [comment]
Technically, yes. You can create a domain which is entirely comprised of one crystal and simulate its deformation response (simply: `neper -T -n 1 ...`).

Operationally, I am skeptical of our ability to properly model the deformation response of a single crystal. The response will be very sensitive to choice of boundary conditions, and I'm afraid that you cannot mitigate their effects. In this way, I would not recommend this path necessarily.

Can you describe the problem further?

**Reply by liangcl123** [reply]
Thank you for your answer. I have two ideas for simulating single superalloys. One approach is to set a domain of a grain that is close to the shape of the workpiece for cyclic loading, and study the influence of orientation by setting different crystal orientations. Another approach is to distinguish between dendrite trunks and interdendritic structures through differences in crystal orientation, considering the influence of carbides, and establishing a two-phase model for simulation to study the differences in deformation responses between dendrite trunks and interdendritic structures.
If the boundary conditions have an impact on the deformation response of a single crystal, should the appropriate boundary conditions be chosen to solve it, or should the constitutive conditions be changed to solve it. Please guide me, thank you!

**Reply by mkasemer** [reply]
Yes, boundary conditions will have an effect on the response. No, I do not suggest a change to constitutive model, which is material dependent and independent of the applied deformation.

The issue with choosing the correct boundary conditions are that the real-life loading is both very complex and transient. Even a rectangular specimen compressed between platens will have complex, transient boundary conditions due primarily to friction, and will again be transient. I don't even know how you would properly characterize what those boundary conditions are over time.

Generally, we make assumptions with boundary conditions. The issue with a single crystal is that it has no "buffer room" between the volume you want to probe and the surfaces at which the BCs are applied. If the BCs cause, for instance, a large stress concentration, it will cause a slip band to form which will likely "take off" across the domain of the sample.

Again, realistically, I'm not entirely sure in what way this can be mitigated, and am skeptical whether single crystals can be properly modeled.

**Reply by rquey** [reply]
@mkasemer With our recent progress on periodic boundary conditions, I think we are actually nearing the point where proper deformation conditions could be applied to single crystals.

Neper 4.9.0 (just released) also allows to generate a periodic mesh of cubic specimens (which oddly was not possible before).

**Reply by liangcl123** [reply]
Thanks for your reply！ Do you mean that in the latest version (4.9.0) it is possible to try to do modeling and simulation of single crystals？ Or keep waiting the newer version. Please guide me, thank you!

**Reply by rquey** [reply]
You can try to follow the [Periodic RVE-type Simulation](https://fepx.info/doc/tutorials/periodic_rve_simulation.html) tutorial, but using only one grain. I'm not saying that it will work or that you will be able to do what you want, though.

Neper commands for a periodic single crystal:
```
neper -T -n 1 -per 1 -o simulation
neper -M simulation.tess
```

**Reply by liangcl123** [reply]
Thank you！ I will try it！
